pkg/workflows/wasm/host/module.go

```diff
@@ -236,6 +242,55 @@ func NewModule(modCfg *ModuleConfig, binary []byte) (*Module, error) {
 		return nil, err
 	}
 
+	err = linker.FuncWrap(
+		"env",
+		"fetch",
+		func(caller *wasmtime.Caller, respptr int32, resplenptr int32, reqptr int32, reqptrlen int32) int32 {
+			b, innerErr := safeMem(caller, reqptr, reqptrlen)
+			if innerErr != nil {
+				logger.Errorf("error calling fetch: %s", innerErr)
+				return 1 //errnumber
+			}
+
+			req := wasm.TargetRequestPayload{}
+			innerErr = json.Unmarshal(b, &req)
```

Comment:
@agparadiso Could we use proto here? We use this elsewhere for our the guest <> host SDK so we should be consistent.